src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq.c

```diff
@@ -2725,17 +2733,18 @@ static ssize_t vmr_system_dtb_read(struct file *filp, struct kobject *kobj,
 	mutex_lock(&xgq->xgq_lock);
 	blob = xgq->xgq_vmr_system_dtb;
 	size = xgq->xgq_vmr_system_dtb_size;
-	mutex_unlock(&xgq->xgq_lock);
 
 	if (off >= size)
 		goto out;
 
 	if (off + count > size)
 		count = size - off;
 	memcpy(buf, blob + off, count);
+	mutex_unlock(&xgq->xgq_lock);
```

Comment:
here, u will unlock twice. it will cause potential undefined behavior or make debugging harder. Please clean up your code.